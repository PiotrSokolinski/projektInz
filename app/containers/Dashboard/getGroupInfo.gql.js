import gql from 'graphql-tag'

export default gql`
  query getGroupInfo($id: ID!) {
    group(id: $id) {
      id
      avatarUrl
      name
      members {
        id
        firstName
        lastName
        avatarUrl
        role
      }
    }
    getTask(id: 0, current: true) {
      id
      name
      description
      createdAt
      status
      priority
      author {
        id
        firstName
        lastName
      }
    }
    getEvent(id: 0, current: true) {
      id
      name
      description
      startDate
      endDate
      author {
        id
      }
      invitedMembers {
        id
        firstName
        lastName
        avatarUrl
      }
    }
  }
`
