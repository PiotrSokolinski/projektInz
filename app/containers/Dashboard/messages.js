/*
 * Dashboard Messages
 *
 * This contains all the text for the Dashboard container.
 */

import { defineMessages } from 'react-intl'

export const scope = 'app.containers.Dashboard'

export default defineMessages({
  membersList: {
    id: `${scope}.membersList`,
    defaultMessage: 'Group Members',
  },
  tasks: {
    id: `${scope}.tasks`,
    defaultMessage: 'Current Group Tasks',
  },
  settings: {
    id: `${scope}.settings`,
    defaultMessage: 'Group Settings',
  },
})
