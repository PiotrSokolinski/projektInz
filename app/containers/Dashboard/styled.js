import styled from 'styled-components'
import { Colors } from 'themes'

export const Container = styled.div`
  display: flex;
  justify-content: space-between;
  height: calc(100vh - 75px);
`
export const TabsContainer = styled.div`
  margin-top: 20px;
  width: 45vw;
  margin-right: 25px;
  .react-tabs__tab-list {
    border-bottom: 1px solid ${Colors.nobel};
  }
  .react-tabs__tab {
    padding: 6px 25px;
  }
  .react-tabs__tab--selected {
    border: 1px solid ${Colors.nobel};
    border-bottom: 1px solid transparent;
    font-weight: bold;
    color: ${Colors.blueRibbon};
  }
`
