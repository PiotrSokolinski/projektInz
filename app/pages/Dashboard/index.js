import React from 'react'
import DashboardContainer from 'containers/Dashboard'
const Dashboard = props => <DashboardContainer {...props} />

export default Dashboard
