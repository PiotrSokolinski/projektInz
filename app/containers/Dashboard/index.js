/**
 *
 * Dashboard
 *
 */

import React from 'react'
// import PropTypes from 'prop-types'
import { FormattedMessage } from 'react-intl'
import messages from './messages'
import * as Styled from './styled'
const Dashboard = () => (
  <Styled.Container>
    <FormattedMessage {...messages.header} />
  </Styled.Container>
)

Dashboard.propTypes = {}

export default Dashboard
