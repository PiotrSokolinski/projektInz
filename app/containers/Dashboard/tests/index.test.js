/**
 *
 * Tests for Dashboard
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react'
import { render } from '@testing-library/react'
import { IntlProvider } from 'react-intl'

import Dashboard from '../index'
import { DEFAULT_LOCALE } from '../../../i18n'

describe('<Dashboard />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error')
    render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <Dashboard />
      </IntlProvider>,
    )
    expect(spy).not.toHaveBeenCalled()
  })

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <Dashboard />
      </IntlProvider>,
    )
    expect(firstChild).toMatchSnapshot()
  })
})
