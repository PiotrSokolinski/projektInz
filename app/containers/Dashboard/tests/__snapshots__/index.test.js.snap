// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> Should render and match the snapshot 1`] = `
<div
  class="styled__Container-sc-1ujewzl-0 FpZuz"
>
  <span>
    This is the Dashboard container!
  </span>
</div>
`;
